analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ClearTextProtocolsAreSensitive.cs

```diff
@@ -61,16 +55,54 @@ public sealed class ClearTextProtocolsAreSensitive : HotspotDiagnosticAnalyzer
             {"clear-text SMTP", "SMTP over SSL/TLS or SMTP with STARTTLS" }
         };
 
+        private readonly List<string> commonlyUsedXmlDomains = new List<string>
+        {
+            "www.w3.org",
+            "xml.apache.org",
+            "schemas.xmlsoap.org",
+            "schemas.openxmlformats.org",
+            "rdfs.org",
+            "purl.org",
+            "xmlns.com",
+            "schemas.google.com",
+            "a9.com",
+            "ns.adobe.com",
+            "ltsc.ieee.org",
+            "docbook.org",
+            "graphml.graphdrawing.org",
+            "json-schema.org"
+        };
+
+        private readonly List<string> commonlyUsedExampleDomains = new List<string> {"example.com", "example.org", "test.com"};
+        private readonly List<string> localhostAddresses = new List<string> {"localhost", "127.0.0.1", "::1"};
+
         private readonly CSharpObjectInitializationTracker objectInitializationTracker =
             new CSharpObjectInitializationTracker(constantValue => constantValue is bool value && value,
                                                   ImmutableArray.Create(KnownType.System_Net_Mail_SmtpClient, KnownType.System_Net_FtpWebRequest),
                                                   propertyName => propertyName == EnableSslName);
 
+        private readonly Regex httpRegex;
+        private readonly Regex ftpRegex;
+        private readonly Regex telnetRegex;
+        private readonly Regex telnetRegexForIdentifier;
+        private readonly Regex validServerRegex;
+
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, EnableSslRule);
 
         public ClearTextProtocolsAreSensitive() : this(AnalyzerConfiguration.Hotspot) { }
 
-        public ClearTextProtocolsAreSensitive(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration) { }
+        public ClearTextProtocolsAreSensitive(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration)
+        {
+            const string allSubdomainsPattern = "([^/]+\\.)?";
+            var domainsList = commonlyUsedXmlDomains.Concat(commonlyUsedExampleDomains.Select(x => allSubdomainsPattern + x)).Concat(localhostAddresses);
+            var validServerPattern = domainsList.JoinStr("|");
```

Comment:
We need to escape the input data (and we have a very useful overload for that)
```suggestion
            var validServerPattern = domainsList.JoinStr("|", x => Regex.Escape(x));
```

Reply:
We cannot do that, since some domains contain regular expressions that should be interpreted as regular expressions

Reply:
We need to encode it selectively then. In more complicated way. Something like
```
var domainsList = localhostAddresses
    .Concat(commonlyUsedXmlDomains)
	.Select(Regex.Escape)
	.Concat(commonlyUsedExampleDomains.Select(x => allSubdomainsPattern + Regex.Escape(x)));
```